### PR TITLE
EasyCert Chrome 58+ compatibility changes

### PR DESCRIPTION
**Pull request by vertexmc**
- subjectAltName added to server certs
- serial number changed from timestamp to guid

### CODE REVIEW

File: synapse/tests/test_lib_certdir.py [hunk 1]
```diff
@@ -39,7 +39,7 @@ def test_certdir_user(self):
     def test_certdir_host(self):
         with self.getCertDir() as cdir:
             cdir.genCaCert('syntest')
-            cdir.genHostCert('visi.vertex.link', signas='syntest')
+            cdir.genHostCert('visi.vertex.link', signas='syntest', sans='DNS:visi.vertex.link,DNS:vertex.link')
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_certdir.py]: do we have a way to load and validate the content of the cert?

File: synapse/tools/easycert.py [hunk 2]
```diff
@@ -67,7 +68,7 @@ def main(argv, outp=None):
             return 0
 
         if opts.server:
-            cdir.genHostCert(opts.name, signas=opts.signas, outp=outp)
+            cdir.genHostCert(opts.name, signas=opts.signas, outp=outp, sans=opts.server_sans)
```
Comment (vEpiphyte) [inline synapse/tools/easycert.py]: whitespace

File: synapse/tools/easycert.py [hunk 3]
```diff
@@ -53,7 +54,7 @@ def main(argv, outp=None):
 
             if opts.ca:
                 cdir.genCaCsr(opts.name, outp=outp)
-                raise Exception('CSR for CA cert not supporte (yet)')
+                raise Exception('CSR for CA cert not supported (yet)')
```
Comment (vEpiphyte) [inline synapse/tools/easycert.py]: :+1:

File: synapse/tests/test_lib_certdir.py [hunk 4]
```diff
@@ -50,7 +50,7 @@ def test_certdir_host(self):
     def test_certdir_hostca(self):
         with self.getCertDir() as cdir:
             cdir.genCaCert('syntest')
-            cdir.genHostCert('visi.vertex.link', signas='syntest')
+            cdir.genHostCert('visi.vertex.link', signas='syntest', sans='DNS:visi.vertex.link')
```
Comment (vEpiphyte) [inline synapse/tests/test_lib_certdir.py]: do we need to set the sans string here? wouldn't  it automatically be set to visi.vertex.link? earlier comment about validating the content applies.

Comment (vertexmc) [inline synapse/tests/test_lib_certdir.py]: Nope, removing.

File: synapse/tools/easycert.py (hunk 2 above)
Comment (vertexmc) [inline synapse/tools/easycert.py]: Is the whitespace incorrect here?

Comment (vEpiphyte) [inline synapse/tools/easycert.py]: its correct - something must've rendered runny with chrome and it looked like there were extraneous spaces.

File: synapse/tests/test_lib_certdir.py (hunk 1 above)
Comment (vertexmc) [inline synapse/tests/test_lib_certdir.py]: I modified the server cert unit test to verify that the SANs were properly added

Comment (vertexmc) [inline synapse/tests/test_lib_certdir.py]: I also added `DNS:vertex.link` to `sans` and the test asserts that it is only added to the cert once.

File: synapse/tests/test_lib_certdir.py [hunk 5]
```diff
@@ -39,13 +39,17 @@ def test_certdir_user(self):
     def test_certdir_host(self):
         with self.getCertDir() as cdir:
             cdir.genCaCert('syntest')
-            cdir.genHostCert('visi.vertex.link', signas='syntest')
+            cdir.genHostCert('visi.vertex.link', signas='syntest', sans='DNS:vertex.link,DNS:visi.vertex.link,DNS:vertex.link')
```
Comment (vertexmc) [inline synapse/tests/test_lib_certdir.py]: My comment was removed when I pushed. I added `DNS:vertex.link` to `sans` twice here and have the test validating the SANs from the loaded cert now (and making sure that `vertex.link` only shows up once).

**Pull request merged**